provider/generic_repository.go

```diff
@@ -47,8 +48,16 @@ func (p *GenericProvider) Gather(user string) ([]GitRepository, error) {
 		return nil, errors.New("This provider requires a object. Example: src-fingerprint -p repository -u 'git@github.com:GitGuardian/gg-shield.git'") // nolint
 	}
 
+	var name string
+	if p.options.RepositoryName != "" {
+		name = p.options.RepositoryName
+	} else {
+		parts := strings.Split(user, "/")
```

Comment:
Makes me think that we should handle the `.git` if any, WDYT ?

Reply:
I've added `name = strings.TrimSuffix(name, ".git")` to remove any `.git` at the end if the URL looks like `git@github.com:GitGuardian/gg-shield.git`. Also the path can be `path/to/project/.git` as it's a valid input for `git clone ...`, I've handled this case by taking the second to last element.